src/main/scala/codacy/metrics/PDepend.scala

```diff
@@ -0,0 +1,144 @@
+package codacy.metrics
+
+import better.files.File
+import codacy.docker.api.{MetricsConfiguration, Source}
+import codacy.docker.api.metrics.{FileMetrics, LineComplexity, MetricsTool}
+import com.codacy.api.dtos.{Language, Languages}
+import com.codacy.docker.api.utils.{CommandResult, CommandRunner}
+
+import scala.util.{Failure, Try}
+import scala.xml.{Node, NodeSeq, XML}
+
+final case class PHPMethod(filename: String, line: Option[Int], complexity: Option[Int])
+
+final case class PHPClass(filename: String, methods: Seq[PHPMethod])
+
+object PDepend extends MetricsTool {
+
+  override def apply(source: Source.Directory,
+                     languageOpt: Option[Language],
+                     files: Option[Set[Source.File]],
+                     options: Map[MetricsConfiguration.Key, MetricsConfiguration.Value]): Try[List[FileMetrics]] = {
+
+    languageOpt match {
+      case Some(language) if language != Languages.PHP =>
+        Failure(new Exception(s"Tried to run PDepend metrics with language: $language. Only PHP supported."))
+      case _ =>
+        val filesToAnalyse: Set[File] = files match {
+          case Some(filesSpecified) =>
+            filesSpecified.map(srcFile => File(source.path) / srcFile.path)
+          case None =>
+            File(source.path).listRecursively().filter(_.isRegularFile).toSet
+        }
+
+        runMetrics(source.path, filesToAnalyse).map(_.toList)
+    }
+  }
+
+  def runMetrics(directory: String, files: Set[File]): Try[Seq[FileMetrics]] = {
+    for {
+      fileOutputStr <- run(directory)
+      xml <- Try(XML.loadString(fileOutputStr))
```

Comment:
Use custom configuration for SAXParser used on codacy/codacy-eslint

Reply:
see https://github.com/codacy/codacy-eslint/blob/native-package/src/main/scala/codacy/XML.scala